### PR TITLE
[RV64_DYNAREC] Added more opcodes for Stardew Valley and some fixes

### PR DESCRIPTION
**Pull request by ksco**

### CODE REVIEW

**Comment by ksco** [comment]
Just found out that the sound of Stardew Valley turned into pure noise after this PR, don't merge yet, I'll check why tomorrow.

File: src/dynarec/rv64/dynarec_rv64_00.c [hunk 1]
```diff
@@ -723,8 +723,35 @@ uintptr_t dynarec64_00(dynarec_rv64_t* dyn, uintptr_t addr, uintptr_t ip, int ni
...
             emit_test32(dyn, ninst, rex, ed, gd, x3, x4, x5);
             break;
+        case 0x86:
+            INST_NAME("(LOCK)XCHG Eb, Gb");
+            nextop = F8;
+            if(MODREG) {
+                GETGD;
+                GETED(0);
+                MV(x1, gd);
+                ANDI(gd, gd, ~0xff);
```
Comment (ptitSeb) [inline src/dynarec/rv64/dynarec_rv64_00.c:733]: the xh part can also be exchanged. I don't think this is handled in this code.

Comment (ksco) [reply src/dynarec/rv64/dynarec_rv64_00.c:733]: Oh yes.

**Comment by ptitSeb** [comment]
Sound is fixed now in SV?

**Comment by ksco** [comment]
No, still no.

**Comment by ksco** [comment]
Sorry for the delay, I have a rest today. Audio is fixed, it's a one-letter mistake.

**Comment by ptitSeb** [comment]
Ah yes. Also, the fact that a RISC Word is 32bits while an x86 Word is 16bits, and DoubleWord is 32bits, doesn't help!

**Comment by ksco** [comment]
Yeah, it's always confusing me.

**Pull request merged**